libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/location/CarLocationProvider.kt

```diff
@@ -13,13 +15,17 @@ import kotlinx.coroutines.delay
 import kotlinx.coroutines.isActive
 import kotlinx.coroutines.withContext
 
-@ExperimentalPreviewMapboxNavigationAPI
-internal class CarAppLocationImpl : CarAppLocation {
+/**
+ * Automatically attaches and detaches from [MapboxNavigationApp] to provide map matched locations.
+ */
+class CarLocationProvider private constructor() : MapboxNavigationObserver, LocationProvider {
```

Comment:
It's used in CarLocationRenderer, am optimizing for the caller rather than the actual CarLocationProvider definition. I'm thinking this class would not be needed if it was done by the NavigationLocationProvider.

```
setLocationProvider(CarLocationProvider.getRegisteredInstance())
setLocationProvider(CarLocationProvider.getRegisteredInstance().navigationLocationProvider)
```

But, it also does not make much sense to expose NavigationLocationProvider when the only need is to access lastLocation. I've added a lastLocation() function and removed the reference to NavigationLocationProvider.

Reply:
ok pushed